src/Event.h

```diff
@@ -0,0 +1,19 @@
+#ifndef _Event_H_
+#define _Event_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef struct Event* EventPtr;
+
+void Events_system_init(void);
+void Event_invoke(EventPtr event);
+uint32_t Event_addListener(EventPtr event, void (*ptr)(void));
+uint32_t Event_removeListener(EventPtr event, void (*ptr)(void));
+void Event_wipeAllListener(EventPtr event);
```

Comment:
This is an unusual function to expose on the event system interface. You might have, say, system-A listening for an event from system-X and also system-B listening for the same system-X event. The value of the event system is decoupling, i.e. systems A, B, and X do not need to know anything about each other. Here you are allowing, for instance, system-A to wipe all listeners and break functionality that system B relies on.  There is likely a better way to design around this behaviour.

Reply:
I've realised it's still useful for resetting an event when 'deleting', but definately shouldnt be public. I've changed it into a static function